folly/Traits.h

```diff
@@ -657,9 +657,7 @@ namespace detail {
 // in order to not prevent all calling code from using it.
 FOLLY_PUSH_WARNING
 FOLLY_GNU_DISABLE_WARNING("-Wsign-compare")
-#if __GNUC_PREREQ(5, 0)
```

Comment:
This currently guards against older versions of clang, too.

```
folly/Traits.h:660:1: error: unknown warning group '-Wbool-compare', ignored [-Werror,-Wunknown-warning-option]
FOLLY_GNU_DISABLE_WARNING("-Wbool-compare")
^
<scratch space>:55:25: note: expanded from here
 GCC diagnostic ignored "-Wbool-compare"
```

Reply:
Instead, should we make this guard a clang specific guard?

Reply:
Yeah. I guess let's change this to guard based on `__clang__`. I don't see any version of Clang which supports this warning.

Can even ask the Clang folks to implement this warning for parity with GCC, and then update the guard here.

Reply:
I just changed it so we only ignore `-Wbool-compare` for GCC only since Clang doesn't support this flag.

I filed https://bugs.llvm.org/show_bug.cgi?id=42130 for asking for Clang to support this flag for symmetry with GCC.